### DISCUSSION TITLE
No posts show after following the Vercel Contentful example guide

### DISCUSSION THREAD

**Discussion by justinzalewski** [vercel/next.js#35955] [Help] [answered]
Hello, I've followed this guide exactly: https://github.com/vercel/next.js/tree/canary/examples/cms-contentful

And no posts show, either on localhost or after deploying to Vercel.

I've double-checked environment variables are set just as the guide specifies.

Any ideas what I can check?

Any help would be much appreciated! Thank you!

**Comment by icyJoseph** [comment]
A little bit of console logging might help. Place one in `getStaticPaths` and `getStaticProps` in `pages/posts/[slug].js`

```ts
export async function getStaticProps({ params, preview = false }) {
  const data = await getPostAndMorePosts(params.slug, preview)

  const ret =  {
    props: {
      preview,
      post: data?.post ?? null,
      morePosts: data?.morePosts ?? null,
    },
  }
 console.log('gSP',ret);
 return ret;
}

export async function getStaticPaths() {
  const allPosts = await getAllPostsWithSlug()
  const ret =  {
    paths: allPosts?.map(({ slug }) => `/posts/${slug}`) ?? [],
    fallback: true,
  }
  console.log("gSPaths", ret);
  return ret;
}
```

When running `next build`, locally, what do you see? Does it even log to anything?

**Reply by justinzalewski** [reply]
@icyJoseph unfortunately, after adding that code and running `npm run dev,` still nothing logged to the console.

**Reply by icyJoseph** [reply]
You did navigate to the relevant pages right? Otherwise the compiler doesn't output it I think. 

I followed that guide, and without any code changes, I see the dummy post and author I created. I can also see them from the landing page.

Here's the output when I build it:

```shell
Page                                                         Size     First Load JS
┌ ● /                                                        3.48 kB        86.8 kB
├   /_app                                                    0 B            71.6 kB
├ ○ /404                                                     192 B          71.8 kB
├ λ /api/exit-preview                                        0 B            71.6 kB
├ λ /api/preview                                             0 B            71.6 kB
└ ● /posts/[slug] (624 ms)                                   6.52 kB        89.8 kB
    └ css/cf121497002c184d.css                               170 B
    ├ /posts/the-brown-fox-jumps-over-the-lazy-dog (440 ms)
    └ /posts/yet-another-dummy-post
+ First Load JS shared by all                                71.6 kB
  ├ chunks/framework-5f4595e5518b5600.js                     42 kB
  ├ chunks/main-681b7976173ab835.js                          28.2 kB
  ├ chunks/pages/_app-81b02de703ea3a5b.js                    493 B
  ├ chunks/webpack-5752944655d749a0.js                       840 B
  └ css/d72df6bc3728bcf5.css                                 2.63 kB
```

As you can see, there's two posts generated at build time, which are the dummy posts I made.


- Did you create `.env.local` out of `.env.example`? 
- Have you followed the guide step by step?
- Are you using the correct API keys?
- Did you create the dummy data on the Contentful CMS dashboard? It didn't let me create authors or posts, until I added a description to the data models (I followed the automated `yarn setup` step)

**Reply by justinzalewski** [reply]
@icyJoseph yes to your four questions :) 

Here's something that might be a clue…when I go to any URL other than the root localhost:3000, I get a 404.

For example, I go to localhost:3000/posts/[slug] (where [slug] is the slug of a post I published in contentful) and it's a 404.

When I do this, I get this in the terminal:
`gSPaths { paths: [], fallback: true }
gSPaths { paths: [], fallback: true }
gSP { props: { preview: false, post: null, morePosts: null } }`

So this makes me think it's an issue of not being correctly connected with Contentful.

However, I've double-checked that I'm using the API keys from here:
<img width="811" alt="Screen Shot 2022-04-08 at 7 58 03 AM" src="https://user-images.githubusercontent.com/2782683/162431164-ed31de5f-da53-412b-8e64-2a42a0517e8b.png">

and I know that the space ID is correct, because it ran the setup correctly (automatically created the content models for me)

**Reply by icyJoseph** [reply]
Could it be that the secret key is incorrect? 

> CONTENTFUL_PREVIEW_SECRET should be any value you want. It must be URL friendly as the dashboard will send it as a query parameter to enable preview mode

- Could you make sure to have only alphanumeric characters on that secret?
- Do you see the posts in the landing page? `locahost:3000` If so, I wonder if the issue are the slugs, maybe they have a character like åiöä´ or something that  (I don't think this is it though, but worth checking out)

if `gsPaths` is returning empty lists for paths, then it is possible that `allPosts` returns `undefined/null`

```ts
export async function getAllPostsWithSlug() {
  const entries = await fetchGraphQL(
    `query {
      postCollection(where: { slug_exists: true }, order: date_DESC) {
        items {
          ${POST_GRAPHQL_FIELDS}
        }
      }
    }`
  )
  console.log({entries});
  return extractPostEntries(entries)
}
```

Try to go to `lib/api.js` and there place a log `console.log({entries})` as shown above, what do you see there?

In the same file there's a function `fetchGraphQL`, control inside it that the env variables have the values you expect, simply log them (that is space_id, and the tokens).

**Reply by justinzalewski** [reply]
@icyJoseph thank you for the quick response!

After adding that console log, the terminal now says this when I go to the intended post URL:
`{ entries: { errors: [ [Object] ] } }
gSPaths { paths: [], fallback: true }
{ entries: { errors: [ [Object] ] } }
gSPaths { paths: [], fallback: true }
gSP { props: { preview: false, post: null, morePosts: null } }`

This weekend, I'm going to try and restart from the guide from scratch (new Contentful space, new Vercel, new repo) and see if that changes anything. 

I'll report back here and let you know how it goes.

**Reply by icyJoseph** [reply]
Awesome! It looks like it's an issue with the data on the Contentful database though. I think it'll work if you re-do it.

**Reply by justinzalewski** [reply]
@icyJoseph after restarting completely from scratch it works! No idea what the issue was the first time. Thanks for your help!

**Comment by justinzalewski** [comment, marked as answer]
Restarting completely from scratch fixed the issue, so marking this answered.
